packages/loader/driver-utils/src/readAndParse.ts

```diff
@@ -31,3 +31,29 @@ export function readAndParseFromBlobs<T>(blobs: {[index: string]: string}, id: s
     const decoded = fromBase64ToUtf8(encoded);
     return JSON.parse(decoded) as T;
 }
+
+/**
+ * Utility to retry the read or fetch until it succeeds in it.
+ * @param api - Method to be retried.
+ */
+export async function readWithRetry<T>(api: () => Promise<T>): Promise<T> {
+    let result: T;
+    try {
+        result = await api();
+    } catch (error) {
+        // If it is not retriable, then just throw the error.
+        if (typeof error !== "object" || error.canRetry !== true) {
+            throw error;
+        }
+        let retryAfter = 0;
```

Comment:
We should reuse logic in DeltaManager to slowly increase timeout from 0 to 8 seconds (or so), and not burn CPU if user is offline.
Also 429s need to be raised as a warning on container. So somehow all of that needs to be connected to container.
I think it can still be separate layer, but likely not separate function, but separate class that has right connections to container to operate. Or maybe even better - it needs to talk to DeltaManager in some way, cause DeltaManager knows already all of that.

Reply:
Added gradual increase of timeout for retrying. Raised throttling warning on container. I think we should keep delta manager separate because that is to manage deltas and should not have anything to do with storage service.